src/webgpu/shader/execution/builtin/select.spec.ts

```diff
@@ -0,0 +1,137 @@
+export const description = `WGSL execution test. Section: Logical built-in functions Function: select`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import {
+  Scalar,
+  TypeVec,
+  TypeBool,
+  TypeF32,
+  TypeI32,
+  TypeU32,
+  f32,
+  i32,
+  u32,
+  False,
+  True,
+  bool,
+  vec2,
+  vec3,
+  vec4,
+} from '../../../util/conversion.js';
+
+import { run } from './builtin.js';
+
+export const g = makeTestGroup(GPUTest);
+
+function makeBool(n: number) {
+  return bool((n & 1) === 1);
+}
+
+type scalarKind = 'b' | 'f' | 'i' | 'u';
+
+const dataType = {
+  b: {
+    type: TypeBool,
+    constructor: makeBool,
```

Comment:
I was thinking of adding these to `ScalarType`, which would be useful for other tests. 

i.e. We could add a `create` field to `ScalarType`:
```ts
/** ScalarType describes the type of WGSL Scalar. */
export class ScalarType {
  readonly kind: ScalarKind; // The named type
  readonly size: number; // In bytes
  readonly create: (value: number) => Scalar; // Constructs a Scalar with the given value
  readonly read: (buf: Uint8Array, offset: number) => Scalar; // reads a scalar from a buffer
```

Then we'd pass these functions to the `ScalarType` constructors:

```ts
export const TypeI32 = new ScalarType('i32', 4, i32, (buf: Uint8Array, offset: number) =>
  i32(new Int32Array(buf.buffer, offset)[0])
);
```

To make this work, I think we'd have to change the `bool(value: boolean)` signature to `bool(value: boolean | number)`, so that bools don't have a different constructor signature.

If you're not keen on changing this, I can take a look at shuffling this to the `ScalarType`.

Reply:
I'll file this as an issue for followup.

Reply:
https://github.com/gpuweb/cts/issues/829